redis/tests/test_basic.rs

```diff
@@ -1184,3 +1184,30 @@ fn test_multi_generics() {
     let _: () = con.rename(999_i64, b"set2").unwrap();
     assert_eq!(con.sunionstore("res", &[b"set1", b"set2"]), Ok(3));
 }
+
+#[test]
+fn test_set_options_with_get() {
+    let ctx = TestContext::new();
+    let mut con = ctx.connection();
+
+    let opts = SetOptions::default().get(true);
+    let data: Option<String> = con.set_options(1, "1", opts).unwrap();
+    assert_eq!(data, None);
+
+    let opts = SetOptions::default().get(true);
+    let data: Option<String> = con.set_options(1, "1", opts).unwrap();
+    assert_eq!(data, Some("1".to_string()));
+}
+
+#[test]
+fn test_options() {
+    let empty = SetOptions::default();
+    assert_eq!(ToRedisArgs::to_redis_args(&empty).len(), 0);
+
+    let opts = SetOptions::default()
+        .conditional_set(ExistenceCheck::NX)
+        .get(true)
+        .with_expiration(SetExpiry::PX(1000));
+
+    assert_args!(&opts, "NX", "GET", "PX", "1000");
```

Comment:
Any reason not to be a bit redundant here and include all variants in the test?

Reply:
Okay, will do